go-controller/pkg/kube/kube.go

```diff
@@ -211,3 +218,49 @@ func (k *Kube) CreateEndpoint(namespace string, ep *kapi.Endpoints) (*kapi.Endpo
 func (k *Kube) Events() kv1core.EventInterface {
 	return k.KClient.CoreV1().Events("")
 }
+
+// AddTaintOnNode set a taint to a node (adds or updates)
+func (k *Kube) SetTaintOnNode(node *kapi.Node, taint kapi.Taint) error {
+	nodeWithTaintMods, _, err := taints.AddOrUpdateTaint(node, &taint)
+	if err != nil {
+		return fmt.Errorf("could not add or update taint of node: %w", err)
+	}
+
+	_, err = k.patchNode(node, nodeWithTaintMods)
+	return err
+}
+
+// RemoveTaintFromNode removes all taints by a key from a node
+func (k *Kube) RemoveTaintFromNode(node *kapi.Node, taint kapi.Taint) error {
+	nodeWithTaintMods, found, err := taints.RemoveTaint(node, &taint)
```

Comment:
This should be in a RetryOnConflict block. If you haven't yet read about k8s ResourceVersion, give this a quick read: https://kubernetes.io/docs/reference/using-api/api-concepts/#resource-versions

Reply:
For an example, see https://github.com/kubernetes/client-go/blob/master/util/retry/util.go#L68

Reply:
@squeed: I moved the RetryOnConflict to `validateGatewayMTU`:
https://github.com/ovn-org/ovn-kubernetes/blob/00dd31620f95fd069306c58a834dd7202689ac46/go-controller/pkg/node/node.go#L551-L563, as the `RemoveTaintFromNode` in #2183 should not retry.